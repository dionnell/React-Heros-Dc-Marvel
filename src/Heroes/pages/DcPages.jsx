import { HeroList } from "../components"

export const DcPages = () => {
  return (
    <>
      <div className="p-3 ps-5 imageDC">
        <h1 className="titulo">Dc Comics</h1>
        <hr />

        <HeroList publisher={'DC Comics'}/>
      </div>
    </>
  )
}
