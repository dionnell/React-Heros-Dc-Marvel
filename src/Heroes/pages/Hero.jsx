import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getHeroById } from "../helpers"
import { useMemo } from "react"



export const Hero = () => {

  const { id } = useParams()

  const hero = useMemo( () => getHeroById(id),[id] ) 

  const navigate = useNavigate()

  const onReturn = () => {
    navigate(-1)
  }

  if ( !hero ){
    return <Navigate to="/marvel" />
  }

  return (
    <div className="container  ps-5">
    <div className="row justify-content-around mt-5">
      <div className="col-4">
        <img 
          src={ `/${ id }.jpg` } 
          alt={hero.superhero}
          className="img-thumbnail animate__bounceIn" 
        />
      </div>
      <div className="col-8 " >
        <h3 className="">{ hero.superhero }</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item"> <b> Alter ego: </b> {hero.alter_ego} </li>
          <li className="list-group-item"> <b> Publisher: </b> {hero.publisher} </li>
          <li className="list-group-item"> <b> First appearance: </b> {hero.first_appearance} </li>
        </ul>

        <h5 className="mt-3" > Characters </h5>
        <p>{hero.characters}</p>

        <button 
          className="btn btn-outline-info"
          onClick={ onReturn }
        >
          Back
        </button>
      </div>


    </div>
    </div>
  )
}
