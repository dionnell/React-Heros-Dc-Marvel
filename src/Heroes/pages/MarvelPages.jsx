import { HeroList } from "../components"

export const MarvelPages = () => {


  return (
    <>
      <div className="p-3 ps-5 imageMarvel">
        <h1 className="titulo">Marvel Comics</h1>
        <hr />

        <HeroList publisher={'Marvel Comics'} />
      </div>
    </>
  )
}
