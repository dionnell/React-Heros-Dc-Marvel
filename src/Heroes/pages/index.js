
export * from './DcPages'
export * from './MarvelPages'
export * from './SearchHero'
export * from './Hero'