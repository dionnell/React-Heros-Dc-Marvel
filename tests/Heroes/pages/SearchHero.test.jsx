import { fireEvent, render,screen } from "@testing-library/react"
import { MemoryRouter, useNavigate } from "react-router-dom"
import { Search } from "../../../src/Heroes/pages/SearchHero"


const mockedUseNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}) )

describe('Pruebas en SearchHero', () => { 
    
    beforeEach( () => jest.clearAllMocks() )

    test('debe de mostrarse correctamente con valores por defecto', () => { 
        
        const {container} = render(
            <MemoryRouter>
                <Search />
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
        //screen.debug()
        
    })

    test('debe de mostrar a batman y el input con el valor del queryString', () => { 
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Search />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        expect(input.value).toBe('batman')

        const img = screen.getByRole('img')
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg')

        const alert = screen.getByLabelText('alert-danger')
        expect(alert.style.display).toBe('none')

        //screen.debug()
        
    })

    test('debe de mostrar un Error si no encuenta el hero', () => { 
        
        render(
            <MemoryRouter initialEntries={['/search?q=robocop']}>
                <Search />
            </MemoryRouter>
        )

        const alert = screen.getByLabelText('alert-danger')
        expect(alert.style.display).toBe('')

        //screen.debug()
        
    })

    test('debe de Llamar el Navigate a la pantalla nueva', () => { 
        
        render(
            <MemoryRouter initialEntries={['/search']}>
                <Search />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        fireEvent.change( input, { target: {name: 'searchText', value:'green' } })
        
        const form = screen.getByRole('form')
        fireEvent.submit(form)
        expect( mockedUseNavigate ).toHaveBeenCalledWith("?q=green")

        //screen.debug()
        
    })


})