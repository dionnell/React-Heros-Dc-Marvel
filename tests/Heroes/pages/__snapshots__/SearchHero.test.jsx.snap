// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchHero debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      Serching
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger"
        style="display: none;"
      >
        hero 
        <b>
           
           
        </b>
         No existing
      </div>
    </div>
  </div>
</div>
`;
